pkg/kubelet/kuberuntime/kuberuntime_container_linux_test.go

```diff
@@ -1259,6 +1261,53 @@ func TestGenerateLinuxContainerResourcesWithSwap(t *testing.T) {
 	}
 }
 
+func TestGenerateUpdatePodSandboxResourcesRequest(t *testing.T) {
```

Comment:
Thanks for adding these tests! Can you turn it into a table-driven test though, and add cases for:
- Best effort pod (no resources defined)
- Guaranteed pod
- Burstable pods that leave some resources unspecified (e.g. only CPU requests, only mem requests)
- With an init container, with a sidecar container
With & without overhead for each case